### ISSUE TITLE
`self` in class is interpreted as a variable

### ISSUE THREAD

**Issue by definitely-not-a-furry** [PyCQA/pylint#8249]
### Bug description

Pylint interpreters `self` in a class as a required variable. I don't know if it's just me or there's actually a bug.

### Configuration

```ini
class MyClass():
    def __init__(self):
        pass

    def MyFunc(self):
        print('something')

MyClass.MyFunc()
```

### Command used

```shell
I use GitHub Codespaces and pylint is preinstalled on VSCode, so no command is used
```

### Pylint output

```shell
No value for argument 'self' in unbound method call `pylint(no-value-for-parameter)`
```

### Expected behavior

I'd expect no errors or warnings

### Pylint version

```shell
pylint 2.15.10
astroid 2.13.3
Python 3.10.4 (main, Jan 25 2023, 00:13:50) [GCC 9.4.0]
```

### OS / Environment

GNU bash, version 5.0.17(1)-release (x86_64-pc-linux-gnu)

### Additional dependencies

anyio==3.6.2
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
attrs==22.2.0
Babel==2.11.0
backcall==0.2.0
beautifulsoup4==4.11.1
bleach==6.0.0
certifi==2022.12.7
cffi==1.15.1
charset-normalizer==3.0.1
colorama==0.4.6
comm==0.1.2
contourpy==1.0.7
cycler==0.11.0
debugpy==1.6.6
decorator==5.1.1
defusedxml==0.7.1
entrypoints==0.4
executing==1.2.0
fastjsonschema==2.16.2
fonttools==4.38.0
fqdn==1.5.1
gitdb==4.0.10
GitPython==3.1.30
idna==3.4
ipykernel==6.20.2
ipython==8.8.0
ipython-genutils==0.2.0
isoduration==20.11.0
jedi==0.18.2
Jinja2==3.1.2
joblib==1.2.0
json5==0.9.11
jsonpointer==2.3
jsonschema==4.17.3
jupyter-events==0.6.3
jupyter-server-mathjax==0.2.6
jupyter_client==7.4.9
jupyter_core==5.1.5
jupyter_server==2.1.0
jupyter_server_terminals==0.4.4
jupyterlab==3.5.3
jupyterlab-git==0.41.0
jupyterlab-pygments==0.2.2
jupyterlab_server==2.19.0
kiwisolver==1.4.4
MarkupSafe==2.1.2
matplotlib==3.6.3
matplotlib-inline==0.1.6
mistune==2.0.4
nbclassic==0.4.8
nbclient==0.7.2
nbconvert==7.2.9
nbdime==3.1.1
nbformat==5.7.3
nest-asyncio==1.5.6
notebook==6.5.2
notebook_shim==0.2.2
numpy==1.24.1
nvidia-cublas-cu11==11.10.3.66
nvidia-cuda-nvrtc-cu11==11.7.99
nvidia-cuda-runtime-cu11==11.7.99
nvidia-cudnn-cu11==8.5.0.96
packaging==23.0
pandas==1.5.3
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.4.0
platformdirs==2.6.2
plotly==5.13.0
prometheus-client==0.16.0
prompt-toolkit==3.0.36
psutil==5.9.4
ptyprocess==0.7.0
pure-eval==0.2.2
pycparser==2.21
Pygments==2.14.0
pyparsing==3.0.9
pyrsistent==0.19.3
python-dateutil==2.8.2
python-json-logger==2.0.4
pytz==2022.7.1
PyYAML==6.0
pyzmq==25.0.0
requests==2.28.2
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
scikit-learn==1.2.1
scipy==1.10.0
seaborn==0.12.2
Send2Trash==1.8.0
six==1.16.0
smmap==5.0.0
sniffio==1.3.0
soupsieve==2.3.2.post1
stack-data==0.6.2
tenacity==8.1.0
terminado==0.17.1
threadpoolctl==3.1.0
tinycss2==1.2.1
tomli==2.0.1
torch==1.13.1
tornado==6.2
traitlets==5.8.1
typing_extensions==4.4.0
uri-template==1.2.0
urllib3==1.26.14
wcwidth==0.2.6
webcolors==1.12
webencodings==0.5.1
websocket-client==1.4.2

**Comment by Pierre-Sassoulas** [comment]
Executing the code gives:
```
Traceback (most recent call last):
  File "/home/pierre/pylint/a.py", line 8, in <module>
    MyClass.MyFunc()
TypeError: MyClass.MyFunc() missing 1 required positional argument: 'self'
```
i.e. pylint is right :)
